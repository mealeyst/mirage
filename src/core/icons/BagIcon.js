import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

const BaseBagIcon = ({ className, count}) => {
  return (
    <svg className={className} version='1.1' viewBox='0 0 20 17'>
      <polygon points='10.5 14.69 10.6 8.79 0.5 8.79 0.5 14.69 2.6 16.49 8.7 16.49 10.5 14.69'/>
      <path d='M3.1,8.39v-1A2.43,2.43,0,0,1,5.5,5a2.43,2.43,0,0,1,2.4,2.4v1Z' />
      {count &&
        <g>
          <circle cx='12.5' cy='7.5' r='7.5'/>
          <text x='0' y='0' className='screenReader'>{`${count} items currently added`}</text>
          <text x='12.3' y='8.5' alignmentBaseline='middle' textAnchor='middle' aria-hidden>{count}</text>
        </g>
      }
    </svg>
  )
}

const BagIcon = styled(BaseBagIcon)`
  width: ${props => props.width};
  > polygon, path {
    fill: none;
    stroke: ${props => props.theme.colors.rocketBlue};
    stroke-linecap:round;
    stroke-linejoin:round;
  }
  > g circle {
    fill: ${props => props.theme.colors.rocketBlue};
  }
  > g text {
    fill: ${props => props.theme.colors.white};
    font-family: ${props => props.theme.fonts.primaryFont};
    font-size: 1rem;
    font-weight: 500;
    letter-spacing: 0;
  }
  .screenReader{
    font-size: 0;
  }
`

BagIcon.propTypes = {
  theme: PropTypes.shape({
    colors: PropTypes.shape({
      rocketBlue: PropTypes.string,
      white: PropTypes.string
    }),
    fonts: PropTypes.shape({
      primaryFont: PropTypes.string
    })
  }),
  width: PropTypes.string
}

BagIcon.propTypes = {
  count: PropTypes.number,
  width: PropTypes.string.isRequired
}

BagIcon.defaultProps = {
  width: '2rem'
}

/** @component */
export default BagIcon
export { BaseBagIcon }
