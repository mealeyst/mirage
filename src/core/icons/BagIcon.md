```js
  <BagIcon count={0}/>
```

## Count Example
```js
  <BagIcon count={20} />
```
