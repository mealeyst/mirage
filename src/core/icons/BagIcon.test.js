import React from 'react'
import { css } from 'styled-components'
import 'jest-styled-components'

import { BagIcon } from 'SRC'
import { BaseBagIcon } from './BagIcon'

const { shallowWithTheme } = global

describe('(Styled Component) BagIcon', () => {
  const createBagIcon = (props) => {
    return shallowWithTheme(<BagIcon {...props} />)
  }

  test('matching the snapshot', () => {
    expect(createBagIcon())
    .toMatchSnapshot()
  })

  test('setting the width', () => {
    const width = '6rem'
    expect(createBagIcon({width: width}))
    .toHaveStyleRule({
      width: width
    })
  })
})

describe('(Component) BaseBagIcon', () => {
  const createBaseBagIcon = (props) => {
    return shallowWithTheme(<BaseBagIcon {...props} />)
  }

  test('matching the snapshot', () => {
    expect(createBaseBagIcon())
    .toMatchSnapshot()
  })

  test('setting the count', () => {
    const count = 4
    expect(
      createBaseBagIcon({count: count})
      .find('.roa-item-count').last().text()
    ).toEqual(count.toString())
    expect(
      createBaseBagIcon({count: count})
      .find('.roa-item-count').first().text()
    ).toEqual(count.toString())
  })
})
