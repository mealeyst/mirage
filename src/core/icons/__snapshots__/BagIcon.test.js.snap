// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) BaseBagIcon matching the snapshot 1`] = `
<div>
  <Styled(Component)
    display="block"
    displayTarget="belowLaptop"
    element="div"
  >
    <svg
      height="18"
      viewBox="0 0 18 18"
      width="18"
    >
      <path
        d="M11.747 6.603H6.253a.577.577 0 0 1-.581-.574V4.903c0-1.815 1.494-3.294 3.329-3.294 1.834 0 3.327 1.479 3.327 3.294v1.126c0 .318-.26.574-.581.574zM12.967 16.89c.163 0 .321-.05.442-.14l2.752-2.064a.528.528 0 0 0 .224-.42l-.016-7.098c0-.31-.3-.563-.666-.563H2.298c-.369 0-.667.252-.667.563l-.016 7.098c0 .16.082.313.224.42L4.59 16.75a.74.74 0 0 0 .443.142h7.934z"
      />
    </svg>
  </Styled(Component)>
  <Styled(Component)
    display="block"
    element="div"
  >
    <svg
      height="20"
      viewBox="0 0 21 20"
      width="21"
    >
      <path
        d="M13.76 7.304H7.217c-.381 0-.692-.288-.692-.646V5.39c0-2.042 1.78-3.706 3.966-3.706 2.184 0 3.963 1.664 3.963 3.706v1.267c0 .358-.31.646-.692.646zM15.214 18.877a.91.91 0 0 0 .526-.159l3.278-2.32a.585.585 0 0 0 .267-.474l-.019-7.985c0-.35-.356-.633-.793-.633H2.506c-.44 0-.795.284-.795.633l-.019 7.985c0 .181.097.353.267.473l3.277 2.321a.918.918 0 0 0 .528.159h9.45z"
      />
    </svg>
  </Styled(Component)>
</div>
`;

exports[`(Styled Component) BagIcon matching the snapshot 1`] = `
.c0 {
  position: relative;
  display: inline-block;
}

.c0 svg {
  width: 1.8rem;
}

.c0 svg > polygon,
.c0 svg path {
  fill: none;
  stroke: #0073D1;
  stroke-linecap: round;
  stroke-linejoin: round;
  stroke-width: 1.5;
}

.c0 .roa-item-count {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: absolute;
  top: -6px;
  right: -7px;
  width: 15px;
  height: 15px;
  border-radius: 50%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #0073D1;
  font-family: "ff-din-web",Helvetica,Arial,sans-serif;
  font-size: 10px;
  font-weight: 500;
  text-align: center;
  color: #FFFFFF;
}

.c0 > g circle {
  fill: #0073D1;
}

.c0 > g text {
  fill: #FFFFFF;
  font-family: "ff-din-web",Helvetica,Arial,sans-serif;
  font-size: .8rem;
  font-weight: 500;
}

.c0 .screenReader {
  font-size: 0;
}

@media (min-width:1024px) {
  .c0 svg {
    width: 2.1rem;
  }

  .c0 svg > g text {
    font-size: .7rem;
  }
}

<BaseBagIcon
  className="c0"
  width={
    Object {
      "desktop": "2.1rem",
      "mobile": "1.8rem",
    }
  }
/>
`;
